### DISCUSSION TITLE
Custom style can't be added to login

### DISCUSSION THREAD

**Discussion by sort72** [laravel/nova-issues#5406] [Ideas & Feature Requests]
- Laravel Version: 9.11
- Nova Version: 4.22.2
- PHP Version: 8.1
- Database Driver & Version: mysql
- Operating System and Version: Windows 11
- Browser type and version: Google Chrome 108.0.5359.125

### Description:

Adding a custom css with `Nova::style('custom', public_path('nova.css'));` in NovaServiceProvider works well for all nova dashboard *except* login/auth routes. This file is not loaded there. I also tried adding this line in AppServiceProvider, didn't work.

### Detailed steps to reproduce the issue on a fresh Nova installation:

1. Add `Nova::style('custom', public_path('nova.css'));` to NovaServiceProvider@boot
2. Create a nova.css file in public path.
3. Add some stuff to the file, example: 
```
.link-default {
    color: yellow;
}
```
4. Check this changed the color for links in Nova, but not in login page.
5. Go to Nova dashboard > Inspect and you will see something like `<link rel="stylesheet" href="/nova-api/styles/custom-admin">` in the `<head>` section. However, this line doesn't exist on Laravel Nova login

**Comment by walaskir** [comment]
Workaround:

First copy view layout.blade.php from Nova to your app
```bash
mkdir -p resources/views/vendor/nova/auth/ && cp vendor/laravel/nova/resources/views/auth/layout.blade.php resources/views/vendor/nova/auth/
```

Then to file `resources/views/vendor/nova/auth/layout.blade.php` add css tag
```html
<link rel="stylesheet" href="{{ asset('css/app.css') }}">
```

Pay attention to the paths which file you copy. Works for me for Nova 3, in version 4 should be similar.

**Comment by mostafaznv** [comment]
I think this is the best solution to apply custom styles to login page

https://github.com/laravel/nova-issues/issues/4789#issuecomment-1244005970

**Comment by ynsbleng** [comment]
It's:

/vendor/laravel/nova/src/Nova.php #857

```
public static function availableStyles(Request $request){
    if (is_null(static::user($request))) {
        return [];
    }
    return static::$styles;
}
```
Not sure why they are making that stupid condition, it prohibit users from registering custom styles for the login as well, the availableStyles method is only called from within /vendor/laravel/nova/resources/views/layout.blade.php, probably removing this condition or at least filtering the styles to keep the user's predefined ones will be better.
